dataflow/src/org/checkerframework/dataflow/cfg/CFGProcessor.java

```diff
@@ -0,0 +1,120 @@
+package org.checkerframework.dataflow.cfg;
+
+import com.sun.source.tree.ClassTree;
+import com.sun.source.tree.CompilationUnitTree;
+import com.sun.source.tree.MethodTree;
+import com.sun.source.util.TreePathScanner;
+import javax.annotation.processing.SupportedAnnotationTypes;
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.TypeElement;
+import org.checkerframework.javacutil.BasicTypeProcessor;
+import org.checkerframework.javacutil.TreeUtils;
+
+@SupportedAnnotationTypes("*")
+public class CFGProcessor extends BasicTypeProcessor {
```

Comment:
This class is a refactor result on original `JavaSource2DOTCFG`. It's a bit un-related to this backward analysis refactor. I will separate it and make another separate PR.